x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/edit_discovery_rule.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPanel,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFormFieldset,
+  EuiSelect,
+  EuiFieldText,
+  EuiFormRow,
+  EuiSuperSelect,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import {
+  Operation,
+  DISCOVERY_RULE_TYPE_ALL,
+  STAGED_DISCOVERY_RULE_ID,
+} from '.';
+
+interface Props {
+  id: string;
+  onChangeOperation: (discoveryItemId: string) => void;
+  operation: string;
+  onChangeType: (discoveryItemId: string) => void;
+  type: string;
+  onChangeProbe: (discoveryItemId: string) => void;
+  probe: string;
+  onCancel: () => void;
+  onSubmit: () => void;
+  operationTypes: Operation[];
+}
+
+export function EditDiscoveryRule({
+  id,
+  onChangeOperation,
+  operation,
+  onChangeType,
+  type,
+  onChangeProbe,
+  probe,
+  onCancel,
+  onSubmit,
+  operationTypes,
+}: Props) {
+  return (
+    <EuiPanel paddingSize="m" hasBorder={true}>
+      <EuiFlexGroup>
+        <EuiFlexItem grow={false}>
+          <EuiFormFieldset legend={{ children: 'Operation' }}>
+            <EuiSelect
+              options={operationTypes.map((item) => ({
+                text: item.operation.label,
+                value: item.operation.value,
+              }))}
+              value={operation}
+              onChange={(e) => {
+                onChangeOperation(e.target.value);
+              }}
+            />
+          </EuiFormFieldset>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormFieldset legend={{ children: 'Type' }}>
+            <EuiFormRow
+              fullWidth
+              helpText={i18n.translate(
+                'xpack.apm.fleetIntegration.apmAgent.runtimeAttachment.editRule.helpText',
+                {
+                  defaultMessage: 'Choose from allowed params',
```

Comment:
Is "params" terminology specific to this that the user is expected to know about or should it be "parameters"?

Reply:
not sure, but I think changing it to `parameters` makes sense.